exchanges/kraken/kraken_wrapper.go

```diff
@@ -674,6 +675,66 @@ func (k *Kraken) GetOrderInfo(orderID string) (order.Detail, error) {
 	return orderDetail, nil
 }
 
+// GetClosedOrderInfo retrieves specified closed order information
+func (k *Kraken) GetClosedOrderInfo(getOrdersRequest *order.GetOrdersRequest) ([]order.Detail, error) {
+	if getOrdersRequest.OrderId == "" {
+		return []order.Detail{}, fmt.Errorf("no order id passed")
+	}
+
+	var orders []order.Detail
+	resp, err := k.QueryOrdersInfo(OrderInfoOptions{
+		Trades: true,
+	}, getOrdersRequest.OrderId)
+	if err != nil {
+		return nil, err
+	}
+
+	a := int64(resp[getOrdersRequest.OrderId].OpenTime)
```

Comment:
We can do a check above and utilise returned var for the subsequent hash lookups below:

```go
respOrderExample, ok := resp[getOrdersRequest.OrderId]
if !ok {
return fmt.Errorf("order %s not found in response", getOrdersRequest.OrderId)
}

responseTime := time.Unix(respOrderExample.OpenTime,0) ... etc

Reply:
Done